service/application/src/test/java/com/solace/maas/ep/event/management/agent/common/messages/ScanCommandMessageTests.java

```diff
@@ -0,0 +1,63 @@
+import com.solace.maas.ep.common.model.ScanType;
+import com.solace.maas.ep.event.management.agent.plugin.mop.MOPMessageType;
+import com.solace.maas.ep.event.management.agent.plugin.mop.MOPProtocol;
+import com.solace.maas.ep.event.management.agent.plugin.mop.MOPUHFlag;
+import com.solace.maas.ep.event.management.agent.subscriber.messageProcessors.EventBrokerResourceConfigTestHelper;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+
+public class ScanCommandMessageTests {
+
+    private void assertMopMessageProperties(ScanCommandMessage scanCommandMessage) {
+        assertNotNull(scanCommandMessage);
+        assertEquals(MOPMessageType.generic, scanCommandMessage.getMopMsgType());
+        assertEquals(MOPProtocol.scanDataControl, scanCommandMessage.getMopProtocol());
+        assertEquals("1", scanCommandMessage.getMopVer());
+        assertEquals(MOPUHFlag.ignore, scanCommandMessage.getMsgUh());
+    }
+
+    @Test
+    void instantiateScanCommandMessage() {
+        ScanCommandMessage scanCommandMessage = new ScanCommandMessage(
+                "messagingServiceId1",
+                "scanId1",
+                List.of(ScanType.SOLACE_ALL),
+                List.of(ScanDestination.EVENT_PORTAL, ScanDestination.FILE_WRITER)
+        );
+        assertMopMessageProperties(scanCommandMessage);
+        assertEquals("messagingServiceId1", scanCommandMessage.getMessagingServiceId());
+        assertEquals("scanId1", scanCommandMessage.getScanId());
+        assertEquals(List.of(ScanType.SOLACE_ALL), scanCommandMessage.getScanTypes());
+        assertEquals(List.of(ScanDestination.EVENT_PORTAL, ScanDestination.FILE_WRITER), scanCommandMessage.getDestinations());
+        assertNull(scanCommandMessage.getResources());
+    }
+
+    @Test
+    void instantiateScanCommandMessageWithResources() {
+        ScanCommandMessage scanCommandMessage = new ScanCommandMessage(
+                "messagingServiceId1",
+                "scanId1",
+                List.of(ScanType.SOLACE_ALL),
+                List.of(ScanDestination.EVENT_PORTAL, ScanDestination.FILE_WRITER),
+                List.of(EventBrokerResourceConfigTestHelper.buildResourceConfiguration(ResourceConfigurationType.SOLACE))
+        );
+        assertMopMessageProperties(scanCommandMessage);
+        assertEquals("messagingServiceId1", scanCommandMessage.getMessagingServiceId());
+        assertEquals("scanId1", scanCommandMessage.getScanId());
+        assertEquals(List.of(ScanType.SOLACE_ALL), scanCommandMessage.getScanTypes());
+        assertEquals(List.of(ScanDestination.EVENT_PORTAL, ScanDestination.FILE_WRITER), scanCommandMessage.getDestinations());
+        assertNotNull(scanCommandMessage.getResources());
+        assertEquals(1, scanCommandMessage.getResources().size());
+    }
```

Comment:
This was not needed on the EMA side (EMA will never instantiate command message with resources) and better suited for ep-core side but you can keep it